src/main/java/com/oneops/secrets/command/SecretDelete.java

```diff
@@ -39,15 +40,24 @@
 
   @Override
   public void exec() {
-
+    Pattern p = Pattern.compile("^y(es)?$");
     try {
-      String in =
-          System.console()
-              .readLine(
-                  warn(
-                      "The delete secret operation is irrevocable. Do you want to proceed (y/n)? "));
-      if (in == null || !in.equalsIgnoreCase("y")) {
-        throw new IllegalStateException("Exiting");
+      if (null != System.console()) {
+        String in =
+            ConsoleImpl.readConsole(
```

Comment:
Again, you need to abstract both System.console and System.in in a single command util. This makes this a little complicated.